src/puppetlabs/puppetdb/http/aggregate_event_counts.clj

```diff
@@ -5,15 +5,16 @@
 
 (defn routes
   [version optional-handlers]
-  (let [param-spec {:required ["summarize_by"]
+  (let [handlers (cons (partial http-q/restrict-query-to-entity "aggregate_event_counts")
+                       optional-handlers)
+        param-spec {:required ["summarize_by"]
                     :optional ["query" "counts_filter" "count_by"
                                "distinct_resources" "distinct_start_time"
                                "distinct_end_time"]}
-        query-route #(apply (partial http-q/query-route :aggregate-event-counts
-                                     version param-spec) %)]
+        query-route #(apply (partial http-q/query-route version param-spec) %)]
```

Comment:
Good thinking, doing that now. Plus, I've made a query-route-from wrapper that automatically puts in the restriction for you, saves on some loc at least :-).